tests/custom_if_prefix/libnetwork_env_var_test.go

```diff
@@ -17,44 +18,48 @@ var _ = Describe("Running plugin with custom ENV", func() {
 			// Run the plugin with custom IFPREFIX
 			RunPlugin("-e CALICO_LIBNETWORK_IFPREFIX=test")
 
+			pool := "test"
+			subnet := "192.169.0.0/16"
 			// Since running the plugin starts etcd, the pool needs to be created after.
-			CreatePool("192.169.0.0/16", false)
+			CreatePool(pool, subnet)
 
 			name := fmt.Sprintf("run%d", rand.Uint32())
-			DockerString(fmt.Sprintf("docker network create %s -d calico --ipam-driver calico-ipam", name))
+			nid := DockerString(fmt.Sprintf("docker network create --driver calico --ipam-driver calico-ipam --subnet %s %s ", subnet, pool))
+			UpdatePool(pool, subnet, nid)
 
 			// Create a container that will just sit in the background
-			DockerString(fmt.Sprintf("docker run --net %s -tid --name %s %s", name, name, os.Getenv("BUSYBOX_IMAGE") ))
+			DockerString(fmt.Sprintf("docker run --net %s -tid --name %s %s", pool, name, os.Getenv("BUSYBOX_IMAGE")))
 
 			// Gather information for assertions
-			docker_endpoint := GetDockerEndpoint(name, name)
-			ip := docker_endpoint.IPAddress
-			mac := docker_endpoint.MacAddress
-			endpoint_id := docker_endpoint.EndpointID
-			interface_name := "cali" + endpoint_id[:mathutils.MinInt(11, len(endpoint_id))]
+			dockerEndpoint := GetDockerEndpoint(name, pool)
+			ip := dockerEndpoint.IPAddress
+			mac := dockerEndpoint.MacAddress
+			endpointID := dockerEndpoint.EndpointID
+			nicName := "cali" + endpointID[:mathutils.MinInt(11, len(endpointID))]
 
 			// Check that the endpoint is created in etcd
-			etcd_endpoint := GetEtcdString(fmt.Sprintf("/calico/v1/host/test/workload/libnetwork/libnetwork/endpoint/%s", endpoint_id))
-			Expect(etcd_endpoint).Should(MatchJSON(fmt.Sprintf(
-				`{"state":"active","name":"%s","active_instance_id": "","mac":"%s","profile_ids":["%s"],"ipv4_nets":["%s/32"],"ipv6_nets":[]}`,
-				interface_name, mac, name, ip)))
+			key := fmt.Sprintf("/calico/resources/v3/projectcalico.org/workloadendpoints/%s/%s-libnetwork-libnetwork-%s", "libnetwork", pool, endpointID)
+			endpointJSON := GetEtcd(key)
+			wep := map[string]interface{}{}
```

Comment:
I think you can use libcalico structs here - e.g. unmarshal wep into an `v3.WorkloadEndpoint` struct.

And, if you're up for it, we could ditch `GetEtcd` altogether and use libcalico to query etcd here. That would clean this code up a lot. However, I don't think we _need_ to do that to get this merged.